### PR TITLE
Genesis

### PR DESCRIPTION
**Pull request by turuslan**
### Description of the Change
[CAR](https://github.com/ipld/go-car) reader.
Refactor CID usage.

### Benefits
Read [genesis from lotus](https://github.com/filecoin-project/lotus/tree/master/build/genesis).

### Possible Drawbacks 
None

### CODE REVIEW

File: core/storage/ipfs/graphsync/impl/merkledag_bridge_impl.cpp [hunk 1]
```diff
@@ -41,7 +38,7 @@ namespace fc::storage::ipfs::graphsync {
     }
 
     // TODO(???): change MerkleDAG service to accept CID instead of bytes
```
Comment (masterjedy) [inline core/storage/ipfs/graphsync/impl/merkledag_bridge_impl.cpp]: Don't forget to specify task

File: test/core/storage/genesis/genesis_test.cpp [hunk 2]
```diff
@@ -0,0 +1,48 @@
...
+using fc::storage::genesis::loadCar;
+using fc::storage::ipfs::InMemoryDatastore;
+
+auto readFile(const std::string &path) {
+  std::ifstream file{path, std::ios::binary | std::ios::ate};
+  EXPECT_TRUE(file.good());
+  Buffer buffer;
+  buffer.resize(file.tellg());
+  file.seekg(0, std::ios::beg);
+  file.read(reinterpret_cast<char *>(buffer.data()), buffer.size());
```
Comment (masterjedy) [inline test/core/storage/genesis/genesis_test.cpp]: It can be done without reinterpret_cast.

File: test/core/storage/genesis/genesis_test.cpp [hunk 3]
```diff
@@ -0,0 +1,48 @@
...
+  return buffer;
+}
+
+TEST(GenesisTest, LoadSuccess) {
```
Comment (masterjedy) [inline test/core/storage/genesis/genesis_test.cpp]: pls add description for tests

File: core/storage/ipfs/graphsync/impl/merkledag_bridge_impl.cpp (hunk 1 above)
Comment (turuslan) [reply core/storage/ipfs/graphsync/impl/merkledag_bridge_impl.cpp]: That's not my todo

**Pull request merged**